src/Polly.Extensions/DependencyInjection/PollyServiceCollectionExtensions.cs

```diff
@@ -135,18 +125,43 @@ public static class PollyServiceCollectionExtensions
         Guard.NotNull(services);
         Guard.NotNull(configure);
 
+        return services.AddResiliencePipelines<TKey>((context) =>
+        {
+            context.AddResiliencePipeline(key, configure);
+        });
+    }
+
+    /// <summary>
+    /// Allows deferred addition of one or more resilience pipelines to the service collection.
+    /// </summary>
+    /// <typeparam name="TKey">The type of the key used to identify the resilience pipelines.</typeparam>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add  resilience pipelines to.</param>
+    /// <param name="configure">An action that allows configuration of resilience pipelines.</param>
+    /// <returns>The updated <see cref="IServiceCollection"/> with the addition configuration added.</returns>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="services"/> or <paramref name="configure"/> is <see langword="null"/>.</exception>
+    /// <remarks>
+    /// This method can be useful if you want to add resilience pipelines as late as possible, e.g.
+    /// to allow other services / configuration to be available before providing the key.
+    /// <para>
+    /// You can retrieve the registered pipeline by resolving the <see cref="ResiliencePipelineProvider{TKey}"/> class from the dependency injection container.
+    /// </para>
+    /// <para>
+    /// This call enables the telemetry for the registered resilience pipeline.
+    /// </para>
+    /// </remarks>
+    public static IServiceCollection AddResiliencePipelines<TKey>(
```

Comment:
Add unit tests for this new API.

Reply:
I did check the unit tests and it looks like the new method was covered by the existing tests for `AddResiliencePipeline` since I refactored them to use `AddResiliencePipelines` internally, per your suggestion.

In this instance, are additional tests still necessary?

Reply:
I would add a test for the new scenario:

- Call `AddResiliencePipelines` and add multiple pipelines
- Ensure that each pipeline is resolvable